libraries/Microsoft.Bot.Builder.Skills/Dialogs/SkillDialog.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Bot.Builder.Skills.Dialogs;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    /// <summary>
+    /// A sample dialog that can wrap remote calls to a skill.
+    /// </summary>
+    /// <remarks>
+    /// The options parameter in <see cref="BeginDialogAsync"/> must be a <see cref="SkillDialogArgs"/> instance
+    /// with the initial parameters for the dialog.
+    /// </remarks>
+    public class SkillDialog : Dialog
+    {
+        private readonly IStatePropertyAccessor<BotFrameworkSkill> _activeSkillProperty;
+        private readonly ConversationState _conversationState;
+        private readonly SkillDialogOptions _dialogOptions;
+
+        public SkillDialog(SkillDialogOptions dialogOptions, ConversationState conversationState)
```

Comment:
>ConversationState conversationState [](start = 61, length = 35)

This can be pulled from dc.Context.TurnState.Get<ConversationState>() which remove need to plumb this through.

Reply:
Is this true outside of Composer? I tried calling turnState.Get<ConversationState> on a standard bot  and it returns null.

Reply:
Talked Carlos and this seems to be set if you are using DialogManager, I am not making that assumption at the moment so this can be used from bots that don't use DialogManager